libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.kt

```diff
@@ -168,142 +210,114 @@ open class LoginActivity : AppCompatActivity(), OAuthWebviewHelperEvents {
+                if (viewModel.singleServerCustomTabActivity) {
+                    finish()
+                } else {
+                    clearWebviewAndShowServerPicker()
+                }
+            }
+        }
 
-    override fun onDestroy() {
-        if (receiverRegistered) {
-            unregisterReceiver(authConfigReceiver)
-            receiverRegistered = false
+        // Take action on server change.
+        viewModel.selectedServer.observe(this) {
+            if (viewModel.singleServerCustomTabActivity) {
+                // Skip fetching authorization and show custom tab immediately.
+                viewModel.reloadWebview()
+                viewModel.loginUrl.value?.let { url ->
+                    loadLoginPageInCustomTab(url, customTabLauncher)
+                }
+            } else {
+                with(SalesforceSDKManager.getInstance()) {
+                    if (useWebServerAuthentication) {
+                        // Fetch well known config and load in custom tab if required.
+                        fetchAuthenticationConfiguration {
+                            if (isBrowserLoginEnabled) {
+                                viewModel.loginUrl.value?.let { url -> loadLoginPageInCustomTab(url, customTabLauncher) }
+                            }
+                        }
+                    }
+                }
+            }
         }
 
-        handleBackBehavior()
-        super.onDestroy()
+        // Let observers know onCreate is complete.
+        EventsObservable.get().notifyEvent(LoginActivityCreateComplete, this)
     }
 
+    override fun onResume() {
+        super.onResume()
+        wasBackgrounded = false
+    }
+
+    override fun onKeyDown(keyCode: Int, event: KeyEvent) =
+        // This allows sub classes to override the behavior by returning false
+        when {
+            fixBackButtonBehavior(keyCode) -> true
+            else -> super.onKeyDown(keyCode, event)
+        }
+
     override fun onNewIntent(intent: Intent) {
         super.onNewIntent(intent)
 
         // If the intent is a callback from Chrome, process it and do nothing else
-        if (isChromeCallback(intent)) {
-            completeAuthFlow(intent)
+        if (isCustomTabAuthFinishedCallback(intent)) {
+            completeAdvAuthFlow(intent)
             return
         }
+    }
 
-        /*
-         * It is important to not reload if we have a custom tab displayed because that also generates
-         * a new code verifier which will break PKCE.
-         */
-        if (!SalesforceSDKManager.getInstance().isBrowserLoginEnabled) {
-            // Reload the login page to ensure the correct login server is displayed in the webview.
-            webviewHelper?.run {
-                loadLoginPage()
-            }
-        }
+    private fun clearWebviewAndShowServerPicker() {
+        viewModel.loginUrl.value = ABOUT_BLANK
+        viewModel.showServerPicker.value = true
     }
```

Comment:
This seemed like a better experience when backing out of a custom tab than seeing the same url loaded in the webview or a blank white activity.